src/textual/_work_decorator.py

```diff
@@ -87,7 +87,7 @@ def work(
     exclusive: bool = False,
     description: str | None = None,
     thread: bool = False,
-) -> Callable[FactoryParamSpec, Worker[ReturnType]] | Decorator:
+) -> Callable[FactoryParamSpec, Worker[ReturnType]] | Decorator[..., ReturnType]:
```

Comment:
This confuses me. Decorator appears to be a type alias and not a generic.

Reply:
`Decorator` is a generic type alias, see the (type variable) `ReturnType` in it:

https://github.com/Textualize/textual/blob/d0f0d76f789a7ed4542b85d196d09875d1fbf86e/src/textual/_work_decorator.py#L23-L31

And my change is to align this usage of `Decorator` in the overload with the previous definition:

https://github.com/Textualize/textual/blob/d0f0d76f789a7ed4542b85d196d09875d1fbf86e/src/textual/_work_decorator.py#L66-L75

Reply:
I don't understand that syntax. It doesn't appear to do anything. And now Mypy is complaining. Are you sure that `Decorator` wouldn't suffice in both places?

Reply:
No, I'm not sure.

With mypy 1.4.1, my change doesn't introduce any new complaints.
mypy complains about line 75, which I didn't touch.

If I change line 75 to use `Decorator` only, mypy stops complaining.

If I use `Decorator` only, pylance complains (in both locations).

What I can do is undo my `Decorator` change and leave it as a follow-up issue.

Reply:
Rolled back this specific change and opened #3510.